### DISCUSSION TITLE
how to create chrome extension in nx?

### DISCUSSION THREAD

**Discussion by HaveF** [nrwl/nx#6643] [General]
Does someone use NX to create chrome extension?

I didn't find any reference(web/repo/issues), and don't know where to start.

It seems like use custom webpack config instead of `@nrwl/react/plugins/webpack` is a reasonable way, but it is a bit complex for me. Or someone has a better idea?

Can someone help me out? Thanks!

**Comment by skyf0xx** [comment]
Best thing is to manually create a directory in NX's apps/my-extension/src

and generate an extension using: https://github.com/extension-js/extension.js

Then at the root package.json, add a dev:myextension, build:myextension under scripts. e.g.

```javascript
{
 "build:chrome-extension": "cd apps/chrome-extension/src && npm run build",
 "dev:chrome-extension": "cd apps/chrome-extension/src && npm run dev",
  "start:chrome-extension": "cd apps/chrome-extension/src && npm run start"
}
```

From your root, you will be able to dev with hot reload using these commands, as if you were using NX and it will work very well with the rest of your mono-repo
